client/src/components/Profile/EducationCard.tsx

```diff
@@ -1,162 +1,154 @@
 import { Typography, List, Input, Button } from 'antd';
+import { ReadOutlined, FileAddOutlined, DeleteOutlined } from '@ant-design/icons';
 import CommonCard from './CommonCard';
-import { GeneralInfo } from 'common/models/profile';
-import { ConfigurableProfilePermissions } from 'common/models/profile';
-import { ChangedPermissionsSettings } from 'pages/profile';
-import { CheckboxChangeEvent } from 'antd/lib/checkbox';
 
 const { Text } = Typography;
 
-import { ReadOutlined, FileAddOutlined, DeleteOutlined } from '@ant-design/icons';
+type UniversityProps = {
+  university: string | null;
+  faculty: string | null;
+  graduationYear: number | string | null;
+};
 
 type Props = {
-  data: GeneralInfo;
+  data: UniversityProps[];
   isEditingModeEnabled: boolean;
-  permissionsSettings?: ConfigurableProfilePermissions;
-  onPermissionsSettingsChange: (event: CheckboxChangeEvent, settings: ChangedPermissionsSettings) => void;
   onProfileSettingsChange: (event: any, path: string) => void;
+  saveProfile: () => void;
 };
 
-class EducationCard extends React.Component<Props> {
-  private filterPermissions = ({ isEducationVisible }: Partial<ConfigurableProfilePermissions>) => ({
-    isEducationVisible,
-  });
+const EducationCard = ({
+  isEditingModeEnabled,
+  data: educationHistory,
+  onProfileSettingsChange,
+  saveProfile,
+}: Props) => {
+  const [universities, setUniversities] = useState(educationHistory);
+  const isDisabled =
+    !!universities.length &&
+    !universities.every(({ university, faculty, graduationYear }) => university && faculty && graduationYear);
+
+  const handleChange = (e: ChangeEvent<HTMLInputElement>, field: keyof UniversityProps, index: number) => {
+    const { value } = e.target;
+
+    setUniversities(prev => {
+      const university = prev[index];
+      university[field] = value;
+      return [...prev];
+    });
+  };
+
+  const handleSave = async () => {
+    await onProfileSettingsChange(universities, 'generalInfo.educationHistory');
+    saveProfile();
+  };
+
+  const handleCancel = () => {
+    setUniversities(educationHistory);
+  };
+
+  const addUniversity = () => {
+    const emptyUniversity = {
+      university: null,
+      faculty: null,
+      graduationYear: null,
+    };
+
+    setUniversities(prev => [...prev, emptyUniversity]);
+  };
 
-  shouldComponentUpdate = (nextProps: Props) =>
-    !isEqual(nextProps.data.educationHistory, this.props.data.educationHistory) ||
-    !isEqual(nextProps.permissionsSettings?.isEducationVisible, this.props.permissionsSettings?.isEducationVisible) ||
-    !isEqual(nextProps.isEditingModeEnabled, this.props.isEditingModeEnabled);
+  const handleDelete = (index: number) => {
+    setUniversities(prev => prev.filter((_, i) => i !== index));
+  };
+
+  const renderSettingsItem = ({ graduationYear, university, faculty }: UniversityProps, index: number) => {
+    const fields = [
+      {
+        label: 'University:',
+        value: university || '',
```

Comment:
It woulde be slightly better to use nullish coalesce here if you are sure about types definitions

Reply:
done